### DISCUSSION TITLE
How to handle TypeError: Cannot read properties of undefined (reading 'constructor') error in bot

### DISCUSSION THREAD

**Discussion by simanye-pav** [telegraf/telegraf#1788] [Q&A]
please help.

I'm making use of the node-mysql2 libraries and telegraf-mysql-session

I have SQL queries and whenever a query is run and some variables are null in that query it triggers this error. This is usually when the session variables are null. AgentID=Null or DealerID=Null. Then the bot doesn't run at all.

**The error originates from the mysql2 connection.js file**

```
C:\inetpub\wwwroot\pavstockallocatebot\node_modules\mysql2\lib\connection.js:538
    if (sql.constructor === Commands.Query) {
            ^
    TypeError: Cannot read properties of undefined (reading 'constructor')
```
**Where it is triggered:**

 ```
query(sql, values, cb) {
    let cmdQuery;
    if (sql.constructor === Commands.Query) {
      cmdQuery = sql;
    } else {
      cmdQuery = Connection.createQuery(sql, values, cb, this.config);
    }
    this._resolveNamedPlaceholders(cmdQuery);
    const rawSql = this.format(cmdQuery.sql, cmdQuery.values !== undefined ? cmdQuery.values : []);
    cmdQuery.sql = rawSql;
    return this.addCommand(cmdQuery);
  }
```
It freezes the bot., and the bot just automatically gets this error whenever I restart it. How can I add an error handler that will handle this error when the bot gets it. Because it's effect the users

**Comment by MKRhere** [comment]
Maybe this is a better issue for https://github.com/hnaderi/telegraf-session-mysql
